src/Tribe/JSON_LD/Abstract.php

```diff
@@ -255,12 +262,27 @@ public function register( $post ) {
 		if ( $this->exists( $id ) ) {
 			return self::$posts[ $id ];
 		}
-
 		self::$posts[ $id ] = get_post( $id );
-
 		return self::$posts[ $id ];
 	}
 
+
+	/**
+	 * Register the current $type to prevent duplicates entries with different $types and IDs
+	 *
+	 * @param $post
+	 */
+	public function register_type( $post ) {
+		$id = Tribe__Main::post_id_helper( $post );
+		if ( isset( self::$types[ $id ] ) && is_array( self::$types ) ) {
+			if ( ! $this->type_exists( $id, $this->type ) ) {
```

Comment:
Avoid complex conditionals like that.
Invert the logic and bail with testable returns.